proof_ics23.go

```diff
@@ -57,12 +80,57 @@ func (t *ImmutableTree) GetNonMembershipProof(key []byte) (*ics23.CommitmentProo
 		}
 	}
 
-	proof := &ics23.CommitmentProof{
-		Proof: &ics23.CommitmentProof_Nonexist{
-			Nonexist: nonexist,
-		},
+	return nonexist, nil
+}
+
+// getNonMembershipProofFast using fast storage
+// invariant: fast storage is enabled
+func (t *ImmutableTree) getNonMembershipProofFast(key []byte) (*ics23.NonExistenceProof, error) {
+	index := 0
+	var prevKey []byte = nil
+	var nextKey []byte = nil
+
+	done := false
+	itr := t.Iterator(nil, nil, true)
```

Comment:
Hrmm, isn't iterating over all of state going to slower? Is there a way we can set a 'smarter' start bound?

Reply:
I think we can do this speedup in a non-db breaking way later, so may also be worth commenting out its invocation in `GetNonMembershipProof` and revisiting after we get the initial release out!